spec/requests/principles_controller_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+RSpec.describe PrinciplesController do
+  before { login_as(user) }
+  let(:principle) { create(:principle) }
+
+  describe 'index' do
+    context 'off season' do
+      include_context 'off season'
+      let(:user) { create(:user) }
+
+      it 'returns forbidden' do
+        get "/groups/#{principle.group.id}/evaluations/#{principle.evaluation.id}/principles"
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context 'application season' do
+      include_context 'application season'
+      let(:user) { principle.evaluation.group.leader.user }
+
+      it 'is successful' do
+        get "/groups/#{principle.group.id}/evaluations/#{principle.evaluation.id}/principles"
+
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'create' do
+    context 'when application season' do
+      include_context 'application season'
+
+      context "when the user is the group leader" do
+        let(:user) { principle.evaluation.group.leader.user }
+
+        it 'creates the principle' do
+          post "/groups/#{principle.group.id}/evaluations/#{principle.evaluation.id}/principles",
+               params: { principle: principle.attributes, format: "javascript" }
+
+          expect(response).to have_http_status(:ok)
```

Comment:
Here I think the expectation does not match the description. What do you think about this?

```suggestion
        it 'creates the principle' do
          request_params = {
            params: { principle: principle.attributes, format: "javascript" }
          }

          expect do
            post "/groups/#{principle.group.id}/evaluations/#{principle.evaluation.id}/principles", **request_params
          end.to change(Principle, :count).by(1)

          expect(response).to have_http_status(:ok)
```